### ISSUE TITLE
[@typescript-eslint/no-unused-vars] False positive no-unused-vars on import in custom.d.ts

### ISSUE THREAD

**Issue by ajmas** [typescript-eslint/typescript-eslint#2287]
I have read the FAQ and have been sent here via the ticket https://github.com/eslint/eslint/issues/13388

**What parser (default, Babel-ESLint, etc.) are you using?**

 default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    'env': {
        'es6': true
    },
    'extends': 'eslint:recommended',
    'globals': {
        'process': true
    },
    'settings': {
        'import/resolver': {
            'node': {
                'extensions': ['.js', '.jsx', '.ts', '.tsx']
            }
        }
    },
    'parser': '@typescript-eslint/parser',
    'parserOptions': {
        'project': 'tsconfig.json',
        'sourceType': 'module'
    },
    'plugins': [
        '@typescript-eslint',
        'eslint-plugin-import'
    ],
    'rules': {
        '@typescript-eslint/indent': ['error', 4],
        '@typescript-eslint/no-require-imports': 'error',
        '@typescript-eslint/quotes': [
            'error',
            'single'
        ],
        'comma-dangle': 'error',
        'curly': 'error',
        'eqeqeq': [
            'error',
            'always'
        ],
        'no-trailing-spaces': 'error',
        'no-var': 'error',
        'prefer-const': 'error',
        'no-console': 'error',
        'no-unused-vars': ['error', { 'args': 'none' }],
        'no-throw-literal': 'error',
        'no-trailing-spaces': 'error',
        'import/no-extraneous-dependencies': 'error',
        'import/no-cycle': 'error',
        'import/no-self-import': 'error',
        'import/no-unresolved': 'error'
    },
    'overrides': [{
        'files': ['*.ts', '*.tsx'],
        'rules': {
            '@typescript-eslint/no-unused-vars': [2, { args: 'none' }],
            'no-undef': 'off'
        }
    }]
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Contents of custom.d.ts file:
```js
/* eslint-disable no-unused-vars */
import { Server } from 'ws';

declare global {
    namespace Express {
        interface Request {
            ws?: Server;
        }
    }
}

export {};
```

```bash
eslint -c .eslintrc.js --ext .ts src
```

**What did you expect to happen?**

I expected to be able to define the type of ws (imported from @types/ws)

**What actually happened? Please include the actual, raw output from ESLint.**

Two different failures:

  - the import is considered unused
  - `eslint-disable no-unused-vars` has no effect
 
**Are you willing to submit a pull request to fix this bug?**

I don't have a fix to offer at this point.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `3.1.0` |
| `TypeScript`                | `3.9.5` |
| `ESLint`                    | `6.8.0` |
| `node`                      | `12.16.2` |
| `npm`                       | `6.13.6` |

**Comment by bradzacher** [comment]
#1856 please use the search.

**Comment by ajmas** [comment]
I had previously, but the lack of activity and response, since April, response gave me the impression the issue had been abandonned.

**Comment by bradzacher** [comment]
nothing in this repo is abandoned.
there is just a small number of maintainers that volunteer their limited free time.
